src/tuntap/macosx/c_tuntap_macosx_obj.cpp

```diff
@@ -23,22 +23,53 @@ size_t c_tuntap_macosx_obj::send_to_tun(const unsigned char *data, size_t size)
 	std::array<boost::asio::const_buffer, 2> buffers;
 	buffers.at(0) = boost::asio::buffer(tun_header);
 	buffers.at(1) = boost::asio::buffer(data, size);
+	try {
+		size_t written_bytes = m_tun_stream.write_some(buffers);
+		if (written_bytes < tun_header.size()) throw std::runtime_error(""); // tun header not written
```

Comment:
std::runtime_error(""); ?

Even though you immediately catch it, maybe we should throw other type that does not take any string/what at all then?

One day someone will copy/paste such code and

![cpp-exception-what](https://user-images.githubusercontent.com/20915731/27411831-afb2acfc-56f0-11e7-9542-5bc1da15ed8a.png)